psh/tools/common.py

```diff
@@ -38,6 +38,25 @@ def create_testdir(p, dirname):
     psh.assert_cmd(p, f'mkdir {dirname}', '', msg)
 
 
+def assert_timestamp_change(p, touched_files, uptimes, dir=''):
+    files = psh.ls(p, dir)
+    for file in files:
+        if file.name in touched_files:
+            uptime = uptimes[file.name]
+            # uptime has to be got before touch
+            # to prevent failed assertion when typing uptime in 00:59 and touch in 01:00
```

Comment:
uh, it would probably better to convert the date to unix timestamp or python datetime object (inside `ls` to be reused between the tests?) and just compare the unix timestamps with some margin (instead of operations on strings).

Please also note that You're only verifying the modification time (access time change is not verified by this test, for that we would need to `stat` the file).